test-node.bash

```diff
@@ -200,17 +212,17 @@ while [[ $# -gt 0 ]]; do
     esac
 done
 
-if $force_init; then
+if ! $skip_build && $force_init; then
```

Comment:
It seems there's currently a lot of disarray in force_build, dev_build_X.. in that way skip_build is reasonable but it adds to the mess instead of removing from it.
I can merge this as I don't have a clear better way - but see if you can come up with something.
I would like the solution to parse things in order, so e.g.:
--init --no-build > will not build, but
--init --no-build --build > will build
(it's a good property for scripts run by other scripts, so you can always change behaviour by adding an argument at the end)

I think some of the mess comes from the duality of build being used for building contracts/scripts vs used to build local nitro.. maybe there should be different args for that.

A final note: external users don't generally use --dev so I don't mind introducing changes that will break usage that does build local nitro.

Reply:
I think I've come up with something that clarifies it, PTAL.